File: Code/Chapter 6/C#/Example 6.9/Form1.cs
#region Using directives

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Windows.Forms;

#endregion

namespace Example_6_9
{
   partial class Form1 : Form
   {
      private BookDataSet books;
      public Form1()
      {
         InitializeComponent();
         books = new BookDataSet();
         books.ReadXml("Books.xml");
         dgBooks.DataSource = books.Books;
      }

      private void btnSumScores_Click(object sender, EventArgs e)
      {
         int sum = 0;
         foreach (BookDataSet.Book book in books.Books)
         {
            foreach (BookDataSet.BookReview review in book.Reviews())
            {
               sum += Convert.ToInt32(review.Rating);
            }
         }
         MessageBox.Show(this, "Score Total: " + sum.ToString());
      }
   }
}

File: Code/Chapter 6/VB.NET/Exercise 6.9/Form1.vb

Public Class Form1
    Private books As BookDataSet
    Private Sub btnSumScores_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnSumScores.Click
        Dim sum As Integer = 0
        Dim Book As BookDataSet.Book
        For Each Book In books.Books
            Dim Review As BookDataSet.BookReview
            For Each Review In Book.Reviews()
                sum += CType(Review.Rating, Integer)
            Next
        Next

        MessageBox.Show(Me, "Score Total: " + sum.ToString())
    End Sub

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        books = New BookDataSet()
        books.ReadXml("Books.xml")
        dgBooks.DataSource = books.Books
    End Sub
End Class
